lib/github-pages-health-check/domain.rb

```diff
@@ -170,7 +174,11 @@ def dns
         return unless valid_domain?
         @dns = Timeout.timeout(TIMEOUT) do
           GitHubPages::HealthCheck.without_warnings do
-            Net::DNS::Resolver.start(absolute_domain).answer unless host.nil?
+            unless host.nil?
+              resolver = Net::DNS::Resolver.new
+              resolver.search(absolute_domain, Net::DNS::A).answer +
+                resolver.search(absolute_domain, Net::DNS::MX).answer
```

Comment:
I don't think it'd be a better approach because our timeouts would then double and so forth. As long as the domain is valid, we need to do a query for MX records, so doing them all at once seems to make sense to me.